diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9abeefb..f80be63 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -197,6 +197,7 @@ add_library(SDL2_image
     IMG_xpm.c
     IMG_xv.c
     IMG_xxx.c
+    IMG_canvas.c
 )
 add_library(SDL2_image::${sdl2_image_export_name} ALIAS SDL2_image)
 target_include_directories(SDL2_image PUBLIC
diff --git a/IMG.c b/IMG.c
index e48ed24..a02ac65 100644
--- a/IMG.c
+++ b/IMG.c
@@ -57,6 +57,7 @@ static struct {
     SDL_Surface *(SDLCALL *load)(SDL_RWops *src);
 } supported[] = {
     /* keep magicless formats first */
+    { "CANVAS", NULL,   IMG_LoadCANVAS_RW },
     { "TGA", NULL,      IMG_LoadTGA_RW },
     { "AVIF",IMG_isAVIF,IMG_LoadAVIF_RW },
     { "CUR", IMG_isCUR, IMG_LoadCUR_RW },
diff --git a/Makefile.am b/Makefile.am
index 0669f2b..1f60a57 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -26,6 +26,7 @@ libSDL2_image_la_SOURCES =	\
 	IMG_svg.c		\
 	IMG_stb.c		\
 	IMG_tga.c		\
+	IMG_canvas.c		\
 	IMG_tif.c		\
 	IMG_xcf.c		\
 	IMG_xpm.c		\
diff --git a/Makefile.in b/Makefile.in
index 7a1afeb..6def5f2 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -153,14 +153,14 @@ LTLIBRARIES = $(lib_LTLIBRARIES)
 am__DEPENDENCIES_1 =
 am__libSDL2_image_la_SOURCES_DIST = IMG.c IMG_avif.c IMG_bmp.c \
 	IMG_gif.c IMG_jpg.c IMG_jxl.c IMG_lbm.c IMG_pcx.c IMG_png.c \
-	IMG_pnm.c IMG_qoi.c IMG_svg.c IMG_stb.c IMG_tga.c IMG_tif.c \
+	IMG_pnm.c IMG_qoi.c IMG_svg.c IMG_stb.c IMG_tga.c IMG_canvas.c IMG_tif.c \
 	IMG_xcf.c IMG_xpm.c IMG_xv.c IMG_webp.c IMG_WIC.c \
 	IMG_ImageIO.m miniz.h nanosvg.h nanosvgrast.h qoi.h \
 	stb_image.h tiny_jpeg.h
 @USE_IMAGEIO_TRUE@am__objects_1 = IMG_ImageIO.lo
 am_libSDL2_image_la_OBJECTS = IMG.lo IMG_avif.lo IMG_bmp.lo IMG_gif.lo \
 	IMG_jpg.lo IMG_jxl.lo IMG_lbm.lo IMG_pcx.lo IMG_png.lo \
-	IMG_pnm.lo IMG_qoi.lo IMG_svg.lo IMG_stb.lo IMG_tga.lo \
+	IMG_pnm.lo IMG_qoi.lo IMG_svg.lo IMG_stb.lo IMG_tga.lo IMG_canvas.lo \
 	IMG_tif.lo IMG_xcf.lo IMG_xpm.lo IMG_xv.lo IMG_webp.lo \
 	IMG_WIC.lo $(am__objects_1)
 libSDL2_image_la_OBJECTS = $(am_libSDL2_image_la_OBJECTS)
@@ -199,7 +199,7 @@ am__depfiles_remade = ./$(DEPDIR)/IMG.Plo ./$(DEPDIR)/IMG_ImageIO.Plo \
 	./$(DEPDIR)/IMG_lbm.Plo ./$(DEPDIR)/IMG_pcx.Plo \
 	./$(DEPDIR)/IMG_png.Plo ./$(DEPDIR)/IMG_pnm.Plo \
 	./$(DEPDIR)/IMG_qoi.Plo ./$(DEPDIR)/IMG_stb.Plo \
-	./$(DEPDIR)/IMG_svg.Plo ./$(DEPDIR)/IMG_tga.Plo \
+	./$(DEPDIR)/IMG_svg.Plo ./$(DEPDIR)/IMG_tga.Plo ./$(DEPDIR)/IMG_canvas.Plo \
 	./$(DEPDIR)/IMG_tif.Plo ./$(DEPDIR)/IMG_webp.Plo \
 	./$(DEPDIR)/IMG_xcf.Plo ./$(DEPDIR)/IMG_xpm.Plo \
 	./$(DEPDIR)/IMG_xv.Plo ./$(DEPDIR)/showanim.Po \
@@ -533,6 +533,7 @@ libSDL2_image_la_SOURCES = \
 	IMG_svg.c		\
 	IMG_stb.c		\
 	IMG_tga.c		\
+	IMG_canvas.c		\
 	IMG_tif.c		\
 	IMG_xcf.c		\
 	IMG_xpm.c		\
@@ -704,6 +705,7 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/IMG_stb.Plo@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/IMG_svg.Plo@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/IMG_tga.Plo@am__quote@ # am--include-marker
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/IMG_canvas.Plo@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/IMG_tif.Plo@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/IMG_webp.Plo@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/IMG_xcf.Plo@am__quote@ # am--include-marker
@@ -1177,6 +1179,7 @@ distclean: distclean-recursive
 	-rm -f ./$(DEPDIR)/IMG_stb.Plo
 	-rm -f ./$(DEPDIR)/IMG_svg.Plo
 	-rm -f ./$(DEPDIR)/IMG_tga.Plo
+	-rm -f ./$(DEPDIR)/IMG_canvas.Plo
 	-rm -f ./$(DEPDIR)/IMG_tif.Plo
 	-rm -f ./$(DEPDIR)/IMG_webp.Plo
 	-rm -f ./$(DEPDIR)/IMG_xcf.Plo
@@ -1248,6 +1251,7 @@ maintainer-clean: maintainer-clean-recursive
 	-rm -f ./$(DEPDIR)/IMG_stb.Plo
 	-rm -f ./$(DEPDIR)/IMG_svg.Plo
 	-rm -f ./$(DEPDIR)/IMG_tga.Plo
+	-rm -f ./$(DEPDIR)/IMG_canvas.Plo
 	-rm -f ./$(DEPDIR)/IMG_tif.Plo
 	-rm -f ./$(DEPDIR)/IMG_webp.Plo
 	-rm -f ./$(DEPDIR)/IMG_xcf.Plo
diff --git a/SDL_image.h b/SDL_image.h
index 3db23b2..20d9009 100644
--- a/SDL_image.h
+++ b/SDL_image.h
@@ -1738,6 +1738,8 @@ extern DECLSPEC SDL_Surface * SDLCALL IMG_LoadQOI_RW(SDL_RWops *src);
  */
 extern DECLSPEC SDL_Surface * SDLCALL IMG_LoadTGA_RW(SDL_RWops *src);
 
+extern DECLSPEC SDL_Surface * SDLCALL IMG_LoadCANVAS_RW(SDL_RWops *src);
+
 /**
  * Load a TIFF image directly.
  *
